lib/DxilPIXPasses/DxilPixReadOutputSig.cpp

```diff
@@ -0,0 +1,128 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// DxilPixEmitMetadata.cpp                                                   //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Emit Dxil resource meta data                                              //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#include "dxc/DXIL/DxilFunctionProps.h"
+#include "dxc/DXIL/DxilModule.h"
+#include "dxc/DXIL/DxilOperations.h"
+#include "dxc/DXIL/DxilUtil.h"
+#include "dxc/DxilPIXPasses/DxilPIXPasses.h"
+#include "dxc/DxilPIXPasses/DxilPIXVirtualRegisters.h"
+#include "dxc/HLSL/DxilGenerationPass.h"
+
+#include "llvm/IR/Module.h"
+
+using namespace llvm;
+using namespace hlsl;
+
+const char *SemanticKindName(DXIL::SemanticKind Kind) {
```

Comment:
Why not use `Semantic::Get(Kind)->GetName()` instead?  You don't have to keep a separate list up to date then.

Reply:
Doh. Didn't find the GetName() function early in this process. Did find it later. Forgot to come back here and clean up.